### ISSUE TITLE
Ensure that all parent containers with dsc[2] information are indexed for Aeon requests

### ISSUE THREAD

**Issue by jrgriffiniii** [pulibrary/pulfalight#160]
The ones with dsc2 description aren't getting indexed.

This is essential to the completion of #159

**Comment by jrgriffiniii** [comment]
This blocked by indexing of the sub-containers within the dsc[2], previously this was just indexed as "Folder 2". This needs to have a link between the PULFALight staging and environment and PULFA 2.0.

**Comment by jrgriffiniii** [comment]
This is a "normal" dsc, for contrast:
```
<dsc type="othertype" othertype="physicalholdings">
         <c level="otherlevel" otherlevel="physicalitem" id="C0040_i1">
            <did>
               <container type="box">1</container>
               <unitid type="barcode">32101040795617</unitid>
               <physloc type="code">rcpxm</physloc>
···············
            </did>
         </c>
      </dsc>
```

**Comment by jrgriffiniii** [comment]
I'm going to resume working with indexing improvements in order to improve the performance for the new "Collection Context" interface work undertaken for https://github.com/pulibrary/pulfalight/issues/170.

**Comment by jrgriffiniii** [comment]
Reviewing the work committed in https://github.com/pulibrary/pulfalight/pull/212, `dsc[2]` elements were actually getting indexed, and I misunderstood the original scope of this issue.

Currently, `components` are indexed within the top-level `collection` Solr Documents. This should document this and clearly outline how to access collection components.

I do not believe that `components` have child `components` within them indexed. That might need to be addressed for this as well.

**Comment by jrgriffiniii** [comment]
https://github.com/pulibrary/pulfalight/commit/7064c7b95d464b9334cec6eb30c28e38b12a0577 seems to be the only proper approach which I could find for recursively processing child components of component elements (`<c>`).

**Comment by tpendragon** [comment]
Recursively nested components are definitely indexed right now. There's a clipboard system that keeps track of parent nodes, and the `//` selector makes sure the selection is arbitrarily deep.

**Comment by jrgriffiniii** [comment]
I was indexing components within the `components` field of the Solr Document with this approach, as I could only find this field populated for top-level collections.

**Comment by jrgriffiniii** [comment]
More generally, I am honestly uncertain if this is even necessary to implement. I might close this issue in favor of another issue within https://github.com/pulibrary/pulfalight/issues/159

**Comment by jrgriffiniii** [comment]
After discussing this with @hackmastera, I am incorrect, and this is still needed in order to ensure that the parent components are accessible for batch requests submitted to the Aeon API.

**Comment by jrgriffiniii** [comment]
The nested components were indeed not being indexed, I will offer a PR off of my work-in-progress branch for this.

**Comment by jrgriffiniii** [comment]
I'm currently finished an approach to getting both child components nested into indexing along with dsc[2] information in https://github.com/pulibrary/pulfalight/commit/8e4b916ff60632c54dbe92e9c2e5119ef7eecbe4. This should probably form the basis also for https://github.com/pulibrary/pulfalight/issues/213

**Comment by jrgriffiniii** [comment]
My initial approach was not structured properly. This is my current understanding of the hierarchy:

- `<container>` elements belong to one or two parents
- in the cases where there are two parents, both are components, but ones lies within the dsc[2]
- dsc[2] `<container>` elements can have multiple children (i. e. multiple `<container>` elements belong to the same dsc[2] component)

The approach that I can instead attempt is to index the dsc[2] containers into a `physicalholdings` field at the level of collections, and for each component which has a `<container>` as a direct child element

**Comment by jrgriffiniii** [comment]
As an aside, I am now finding that the Lando-based Solr installation *does* respond to queries for nested documents as expected with the query parameter `fl=*,[child]`.

**Comment by jrgriffiniii** [comment]
This should build upon the refactoring I am looking to introduce with https://github.com/pulibrary/pulfalight/compare/issues-247-jrgriffiniii-index-components-depth-refactor

**Comment by jrgriffiniii** [comment]
This is no longer needed given that we are going to be indexing instead from the ASpace-exported EADs.
